### PR TITLE
Add externs for Angular 1.5 component

### PR DESCRIPTION
**Pull request by gberaudo**
As described in https://docs.angularjs.org/api/ng/provider/$compileProvider#component.

[Review on Reviewable](https://reviewable.io/reviews/google/closure-compiler/1547)

### CODE REVIEW

**Comment by dimvar** [comment]
@mprobst Can you take a look at this?

File: contrib/externs/angular-1.5.js [hunk 1]
```diff
@@ -717,6 +735,15 @@ angular.Module = function() {};
 angular.Module.prototype.animation = function(name, animationFactory) {};
 
 /**
+ * Intended to be called with a name string and a component definition.
+ *
+ * @param {string} name
+ * @param {Object<string, *>} componentDefinition
```
Comment (mprobst) [inline contrib/externs/angular-1.5.js]: Shouldn't this reference `!angular.Component` ?

**Comment by mprobst** [comment]
One nit, LGTM otherwise.

File: contrib/externs/angular-1.5.js (hunk 1 above)
Comment (gberaudo) [inline contrib/externs/angular-1.5.js]: Unfortunately, using `!angular.Component` would require a cast in user code to handle [optional properties](https://groups.google.com/forum/#!topic/closure-compiler-discuss/3qhcDfa9uX4) or to handle `"custom properties"` (`$... – {function()=} – additional annotations to provide to the directive factory function.`).

Directives are handled with a `angular.Injectable=` param which is also a loose type.

I added the missing `!`; if you still prefer `!angular.Component` I can update the PR.

**Comment by mprobst** [comment]
Shouldn't the `typedef` with a record type allow passing objects that are lacking the optional fields?

```js
/** @typedef {{foo: string, bar: (number|undefined)}} */
var Foo;

/** @param {!Foo} foo */
function useFoo(foo) {}

useFoo({foo: 'asd'}); // no 'bar'
```

**Comment by gberaudo** [comment]
Thank you @mprobst, I updated the PR.

With a compiler from October and the `--jscomp_error "*"` flag, I had an `initializing variable` error.
With a version from December it now passes, which is very great!

Is it a new feature?

**Comment by gberaudo** [comment]
I have [a pile of Travis builds](https://travis-ci.org/google/closure-compiler/pull_requests) stuck in `creation mode`. I cannot cancel jobs; could someone try to cancel the older ones and restart the last one?

**Comment by Dominator008** [comment]
@gberaudo Travis just takes a while during US business hours... Hang in there :)

**Comment by gberaudo** [comment]
OK, will try to be patient ^^.

**Comment by mprobst** [comment]
@gberaudo: my understanding was that this was always the intended behaviour, but this aspect of Closure Compiler seems like a less well-trodden path.

**Pull request merged**